service/src/integrationTest/java/uk/gov/hmcts/reform/fpl/controllers/UploadTranslationsSubmittedControllerTest.java

```diff
@@ -0,0 +1,284 @@
+
+    @Captor
+    private ArgumentCaptor<LetterWithPdfsRequest> printRequest;
+
+    @MockBean
+    private SendLetterApi sendLetterApi;
+
+    @MockBean
+    private NotificationClient notificationClient;
+
+    @MockBean
+    private CoreCaseDataService coreCaseDataService;
+
+    @MockBean
+    private UploadDocumentService uploadDocumentService;
+
+    @MockBean
+    private DocumentDownloadService documentDownloadService;
+
+    @MockBean
+    private DocmosisCoverDocumentsService documentService;
+
+    @SpyBean
+    private EventService eventPublisher;
+
+    UploadTranslationsSubmittedControllerTest() {
+        super("upload-translations");
+    }
+
+    @BeforeEach
+    void mocks() {
+        givenFplService();
+        when(documentDownloadService.downloadDocument(anyString()))
+            .thenReturn(ORDER_BINARY);
+        when(uploadDocumentService.uploadPDF(ORDER_BINARY, ORDER.getFilename()))
+            .thenReturn(ORDER_DOCUMENT);
+
+        when(documentService.createCoverDocuments(any(), any(), eq(REPRESENTATIVE_POST.getValue())))
+            .thenReturn(DocmosisDocument.builder().bytes(COVERSHEET_REPRESENTATIVE_BINARY).build());
+        when(uploadDocumentService.uploadPDF(COVERSHEET_REPRESENTATIVE_BINARY, "Coversheet.pdf"))
+            .thenReturn(COVERSHEET_REPRESENTATIVE);
+
+        when(documentService.createCoverDocuments(any(), any(), eq(RESPONDENT_NOT_REPRESENTED.getParty())))
+            .thenReturn(DocmosisDocument.builder().bytes(COVERSHEET_RESPONDENT_BINARY).build());
+        when(uploadDocumentService.uploadPDF(COVERSHEET_RESPONDENT_BINARY, "Coversheet.pdf"))
+            .thenReturn(COVERSHEET_RESPONDENT);
+
+        when(sendLetterApi.sendLetter(any(), any(LetterWithPdfsRequest.class)))
+            .thenReturn(new SendLetterResponse(LETTER_1_ID), new SendLetterResponse(LETTER_2_ID));
+    }
+
+    @Test
+    @SuppressWarnings("unchecked")
+    void shouldSendTranslatedNotificationToLocalAuthorityWhenTranslatedOrder() {
+        CaseData caseData = caseData();
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(asCaseDetails(caseData))
+            .build();
+
+        postSubmittedEvent(request);
+
+        checkUntil(() -> verify(notificationClient, timeout(ASYNC_METHOD_CALL_TIMEOUT)).sendEmail(
+            eq(ITEM_TRANSLATED_NOTIFICATION_TEMPLATE),
+            eq(LOCAL_AUTHORITY_1_INBOX), eqJson(NOTIFICATION_PARAMETERS),
+            eq(NOTIFICATION_REFERENCE)
+        ));
+    }
+
+    @Test
+    void shouldSendOrdersByPostWhenOrderTranslated() {
+        CaseData caseData = caseData();
+        CallbackRequest request = CallbackRequest.builder()
+            .caseDetails(asCaseDetails(caseData))
+            .build();
+
+        postSubmittedEvent(request);
+
+        checkUntil(() -> {
+            verify(sendLetterApi, timeout(ASYNC_METHOD_CALL_TIMEOUT).times(2)).sendLetter(eq(SERVICE_AUTH_TOKEN),
```

Comment:
not sure if the timeout is required as I think that is what the checkUntil provides, might be wrong though